api/envoy/config/route/v3/route_components.proto

```diff
@@ -542,6 +542,33 @@ message RouteMatch {
     //
     // Expect the value to not contain ``?`` or ``#`` and not to end in ``/``
     string path_separated_prefix = 14 [(validate.rules).string = {pattern: "^[^?#]+[^?#/]$"}];
+
+    // If specified, the route is a template match rule meaning that the
+    // ``:path`` header (without the query string) must match the given
+    // ``path_template_match`` pattern.
+    //
+    // Path template matching types:
+    //
+    // * ``*`` : Matches a single path component, up to the next path separator: /
+    //
+    // * ``**`` : Matches zero or more path segments. If present, must be the last operator.
+    //
+    // * ``{name} or {name=*}`` :  A named variable matching one path segment up to the next path separator: /.
+    //
+    // * ``{name=videos/*}`` : A named variable matching more than one path segment.
+    //      The path component matching videos/* is captured as the named variable.
+    //
+    // * ``{name=**}`` : A named variable matching zero or more path segments.
+    //
+    //
+    // For example:
+    //
+    // * ``/videos/*/*/*.m4s`` would match ``videos/123414/hls/1080p5000_00001.m4s``
+    //
+    // * ``/videos/{file}`` would match ``/videos/1080p5000_00001.m4s``
+    //
+    // * ``/**.mpd`` would match ``/content/123/india/dash/55/manifest.mpd``
+    string path_template_match = 15 [(validate.rules).string = {min_len: 1}];
```

Comment:
pls add max_len validation as well, the g-doc says max length 256, I think that's very reasonable.

Reply:
Added